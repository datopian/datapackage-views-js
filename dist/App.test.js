"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _react2 = require("@testing-library/react");

var _App = _interopRequireDefault(require("./App"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var datapackage = {
  views: [{
    'id': 1,
    'title': 'GDP',
    'resources': ['gdp'],
    'specType': 'table'
  }]
};
it('renders spinner', function () {
  var _render = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: datapackage,
    loading: true
  })),
      container = _render.container;

  expect(container.querySelector('svg')).toMatchSnapshot();
});
it('renders error message when data is unavailable', function () {
  var copyOfDp = JSON.parse(JSON.stringify(datapackage));
  copyOfDp.views[0].resources[0] = {
    name: 'gdp',
    unavailable: true
  };

  var _render2 = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: copyOfDp
  })),
      getByText = _render2.getByText;

  expect(getByText('Data view unavailable.')).toBeInTheDocument();
});
it('renders a preview table when data is loaded', function () {
  var copyOfDp = JSON.parse(JSON.stringify(datapackage));
  copyOfDp.views[0].resources[0] = {
    name: 'gdp',
    data: [{
      a: 1,
      b: 2
    }, {
      a: 3,
      b: 4
    }]
  };

  var _render3 = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: copyOfDp
  })),
      container = _render3.container;

  expect(container.querySelector('table.htCore')).toMatchSnapshot();
});
it('renders a preview table with custom headers', function () {
  var copyOfDp = JSON.parse(JSON.stringify(datapackage));
  copyOfDp.views[0].resources[0] = {
    name: 'gdp',
    data: [{
      a: 1,
      b: 2
    }, {
      a: 3,
      b: 4
    }],
    schema: {
      fields: [{
        name: 'a',
        title: 'custom-a'
      }, {
        name: 'b',
        title: 'custom-b'
      }]
    }
  };

  var _render4 = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: copyOfDp
  })),
      container = _render4.container;

  expect(container.querySelector('table.htCore')).toMatchSnapshot();
});
it('renders a Map for geojson resources', function () {
  var copyOfDp = JSON.parse(JSON.stringify(datapackage));
  copyOfDp.views[0].specType = 'map';
  copyOfDp.views[0].resources[0] = {
    name: 'map',
    data: {
      "type": "Feature",
      "geometry": {
        "type": "Point",
        "coordinates": [125.6, 10.1]
      },
      "properties": {
        "name": "Dinagat Islands"
      }
    }
  };

  var _render5 = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: copyOfDp
  })),
      container = _render5.container;

  expect(container.firstChild).toMatchSnapshot();
});
it('renders a Document for PDF resources', function () {
  var copyOfDp = JSON.parse(JSON.stringify(datapackage));
  copyOfDp.views[0].specType = 'document';
  copyOfDp.views[0].resources[0] = {
    name: 'document',
    format: 'pdf',
    path: 'some-path-to-pdf'
  };

  var _render6 = (0, _react2.render)(_react.default.createElement(_App.default, {
    datapackage: copyOfDp
  })),
      container = _render6.container;

  expect(container.firstChild).toMatchSnapshot();
});