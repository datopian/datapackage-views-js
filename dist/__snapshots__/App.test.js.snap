// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Document for PDF resources 1`] = `
<div
  class="App"
>
  <div
    class="container m-24"
  >
    <div>
      <span>
        Loading...
      </span>
      <canvas />
    </div>
  </div>
</div>
`;

exports[`renders a Map for geojson resources 1`] = `
<div
  class="App"
>
  <div
    class="container m-24"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="width: 100%; height: 450px; position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://c.tile.openstreetmap.org/10/869/483.png"
                style="width: 256px; height: 256px; left: -67px; top: -31px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        >
          <img
            alt=""
            class="leaflet-marker-shadow leaflet-zoom-hide"
            src="marker-shadow.png"
            style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
          />
        </div>
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt=""
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            src="marker-icon.png"
            style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
             | © 
            <a
              href="http://osm.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a preview table when data is loaded 1`] = `
<table
  class="htCore"
>
  <colgroup>
    <col
      style="width: 47px;"
    />
  </colgroup>
  <thead>
    <tr>
      <th
        class=""
        title="a"
      >
        <div
          class="relative"
        >
          <span
            class="colHeader columnSorting sortAction"
          >
            a
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="htDimmed"
      >
        1
      </td>
    </tr>
    <tr>
      <td
        class="htDimmed"
      >
        3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders a preview table with custom headers 1`] = `
<table
  class="htCore"
>
  <colgroup>
    <col
      style="width: 47px;"
    />
  </colgroup>
  <thead>
    <tr>
      <th
        class=""
        title="custom-a"
      >
        <div
          class="relative"
        >
          <span
            class="colHeader columnSorting sortAction"
          >
            custom-a
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="htLeft htDimmed"
      >
        1
      </td>
    </tr>
    <tr>
      <td
        class="htLeft htDimmed"
      >
        3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders spinner 1`] = `
<svg
  fill="#D3D3D3"
  height="50"
  viewBox="0 0 105 105"
  width="50"
>
  <circle
    cx="12.5"
    cy="12.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="12.5"
    cy="52.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="100ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="12.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="300ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="52.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="600ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="12.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="800ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="52.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="400ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="12.5"
    cy="92.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="700ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="92.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="500ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="92.5"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="200ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
</svg>
`;
